utils/k8s.go

```diff
@@ -0,0 +1,178 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"github.com/digitalocean/godo"
+	log "github.com/sirupsen/logrus"
+	v12 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+	"strings"
+	"time"
+)
+
+type FullNode struct {
+	ClusterId string
+	PoolId string
+	NodeId string
+}
+
+func AuthenticateInCluster() (*kubernetes.Clientset, dynamic.Interface, error) {
+	// creates the in-cluster config
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get client config: %v", err)
+	}
+
+
+	// creates the clientset
+	clientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate client set: %v", err)
+	}
+
+
+	// creates dynClient for kube watching
+	dynClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate dynamic client : %v", err)
+	}
+
+	return clientSet, dynClient, nil
+}
+
+func AuthenticateOutOfCluster() (*kubernetes.Clientset, dynamic.Interface, error) {
+	kubeconfig := os.Getenv("KUBECONFIG")
+
+	// use the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get client config: %v", err)
+	}
+
+	// create the clientset
+	clientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate client set: %v", err)
+	}
+
+	// creates dynClient for kube watching
+	dynClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate dynamic client : %v", err)
+	}
+
+	return clientSet, dynClient, nil
+}
+
+func GetKubeStuckNodesInfos(clientSet *kubernetes.Clientset) []FullNode {
+	nodes, err := clientSet.CoreV1().Nodes().List(context.TODO(), v1.ListOptions{Limit: int64(1000)})
+
+	if err != nil {
+		log.Errorf("Can't list kubernetes nodes: %s", err.Error())
+	}
+
+	stuckNodes := getStuckNodes(nodes)
+
+	if len(stuckNodes) == len(nodes.Items) {
+		log.Debug("Cluster isn't ready yet.")
+		return []FullNode{}
+	}
+
+	return stuckNodes
+}
+
+/**
+* Using three conditions to check if node is stock : node Ready status is at "unknown", it's not Unschedulable and it was
+* created more than five minutes ago.
+ */
+func checkNode(node v12.Node) FullNode{
+	var fullNode FullNode
+
+	for _, condition := range node.Status.Conditions {
+		if strings.EqualFold(fmt.Sprint(condition.Type), "Ready") &&
+			strings.EqualFold(fmt.Sprint(condition.Status), "Unknown") &&
+			!node.Spec.Unschedulable &&
+			node.CreationTimestamp.Add(time.Minute *5).Before(time.Now()) {
```

Comment:
can you please make this field configurable (with env vars for example). if we see 5 min is not enough/too much, we'll easily be able to update it

Reply:
done